### DISCUSSION TITLE
One to many and many to many relation on same table

### DISCUSSION THREAD

**Discussion by CarloAndreoli1994** [prisma/prisma#6542] [Q&A] [answered]
Hello everyone!

I have two table:

```prisma
model User {
  id                              Int                          @id @default(autoincrement())
  firstname                 String
  lastname                  String?
  email                        String?
  news                        News[]
  subscriptions          News[]                  @relation("NewsToUserManyToMany")
}

model News {
  id                            Int                       @id @default(autoincrement())
  title                        String
  creationDate         DateTime           @default(now())
  author                    User                    @relation(fields: [authorId], references: [id])
  targetUsersId        User[]                 @relation("NewsToUserManyToMany")

  @@index([authorId], name: "userId")
}
```

I have two relation between them, the first is a one to many with the ```authorId``` referred to the User and I'm trying to set a many to many relation to define the subscription of the users to the different news.

Every time I try to query the many to many table for defining (or selecting) the targetUsers I had this error:

```
{
    "message": "database request validation failed",
    "details": {
        "table": "_NewsToUserManyToMany"
    }
}

The table `_NewsToUserManyToMany` does not exist in the current database.'
```

I think the problem is the one to many relation because every other many to many are working good without any other relation to the linked table.

Does anyone can help on this?

Thank you.

**Comment by ryands17** [comment]
@CarloAndreoli1994 👋 

Could you share your query as well?

**Reply by CarloAndreoli1994** [reply]
Simply trying to find all entries including the relation

```ts
prisma.news.findMany({
 include: {
    targetUsersId: true
 }
}

**Reply by ryands17** [reply, marked as answer]
This works for me with the following schema:

```prisma
model User {
  id            Int     @id @default(autoincrement())
  firstname     String
  lastname      String?
  email         String?
  news          News[]
  subscriptions News[]  @relation("NewsToUserManyToMany")
}

model News {
  id            Int      @id @default(autoincrement())
  title         String
  creationDate  DateTime @default(now())
  author        User     @relation(fields: [authorId], references: [id])
  targetUsersId User[]   @relation("NewsToUserManyToMany")
  authorId      Int

  @@index([authorId], name: "userId")
}
```

I tried it with the latest version of `prisma` and `@prisma/client` i.e. **2.20.1**. Your schema needs an `authorId` as well. Could you try it on the latest version and check?

**Reply by CarloAndreoli1994** [reply]
Hi ryands, thanks for your support.
I tried to follow your advice and I've updated my prisma and prismaClient to the latest version but I still get the same error (I have the authorId in the schema, forgot to insert it here).

I found a solution calling the relation table exactly as the schema requires '_NewsToUserManyToMany' and seems to work this way

**Reply by ryands17** [reply]
That's not required, but it's fine as long as it's working. It would be great if you could provide me with a sample reproduction so that I can check.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
